docs/chain/did.md

```diff
@@ -0,0 +1,73 @@
+# DID 
+
+## What is a DID?
+
+In KILT a DID is a decentralized identifier that the user owns and controls. It consists of a unique set of keys that can be used for different operations on the blockchain. For an in-depth explanation see the [KILT DID spec](https://github.com/KILTprotocol/kilt-did-driver/blob/master/docs/did-spec/spec.md).
+
+A DID may be a "light" DID, which is not stored on-chain, or a "full" on-chain DID. A light DID is issued by default, with the keys stored locally on your device. By upgrading this to a full DID registered on the blockchain, all the keys associated with it can be retrieved from the KILT blockchain storage.
+
+A full DID can then be used to perform certain on-chain actions which include:
+
+* Writing CTypes to the chain
+* Writing attestations to the chain
+* Setting delegations
+* Doing key rotations on the DID keys
+
+## Registering a DID
+
+A full DID is needed if the user wants to become an attester or wants to setup delegations. A full DID also allows the user to embed a list of URLs, known as service endpoints, into the DID document so that it can be retrieved from the chain.
+
+To create a full DID the user first has to create a set of keys and service endpoints:
+
+* one authentication key for signing extrinsics from your DID
+* zero or more key agreement keys for encrypting messages that are sent to you
+* one attestation key for signing attestations (optional)
+* one delegation key for authorizing delegations (optional)
+* service endpoints that point to external hostings for others to find (optional)
+
+With these they are ready to write the DID to the KILT blockchain.
+
+The user then has to create the `did::create` extrinsic and sign it with any KILT account that has enough funding to pay both the transaction fees and the  DID deposit. The extrinsic consists of  
+
+* The `DidCreationDetails` object containing keys and service endpoints
+* The `DidSignature` which is a signature using your authentication key over the scale encoded `DidCreationDetails` from above
+* A regular signature authenticating the sender of the extrinsic
+
+Once this extrinsic is submitted and executed, the DID is written to the chain.
```

Comment:
I would like to see a bit more details here. What I miss here is how the `DidCreationDetails` look like, what purpose the `submitter` serves, that the submitter and DID owner can be two different parties and what consequences that would have.

Reply:
The purpose of the submitter wasn't made clear i think? This doc is a more low level explanation of the DIDs, so I think we should explain the details here too.
The submitter is included in the DidCreationDetails, so that the user can limit who pays the deposit to this specific account. Otherwise some other account might pick up the `DidCreationDetails` and become the deposit owner with the power to delete the DID later.